ops/lost-in-translation.js

```diff
@@ -0,0 +1,163 @@
+ * to one or many locale files (usually the english locale (en.json) or to their own locale
+ * file (nl.json), or both).
+ *
+ * But they probably haven't added translations to all the other locales, because they don't
+ * speak the other languages. And this is the problem, because now we need find people to help
+ * translate the added keys to all the other locales.
+ *
+ * So what we do here is simply to find *all* the english translation keys across the entire
+ * project. That means looking at all the english translations keys in all the locale files
+ * since the start of the project across all branches and PRs. We throw them into a set that
+ * we use in the next step.
+ */
+const allLocales = retrieveAllLocales('app/locales/');
+const allEnglishTranslationKeys = new Set([]);
+for (const locale of allLocales) {
+  const filePath = `app/locales/${locale}.json`;
+  for (const commitHash of findCommitHashesForFile(filePath)) {
+    const translation = retrieveJSONForFileAtCommitHash(filePath, commitHash);
+    for (const translationKey of Object.keys(translation)) {
+      allEnglishTranslationKeys.add(normalizeTranslationKey(translationKey));
+    }
+  }
+}
+
+/**
+ * Step 2: For each locale file check if there are any missing english translation keys
+ *         across all branches and PRs and all commits/changes.
+ *
+ * If there's a missing english translation key, we know that we're most likely missing
+ * a translation for that locale. So what we need to do is to translate it, or get help
+ * to translate it.
+ * 
+ * We throw it into a dictionary where the key is the locale and the value is a set of 
+ * missing translations from english to that locale.
+ */
+const translationKeysByLocale = {};
+for (const locale of allLocales) {
+  const filePath = `app/locales/${locale}.json`;
+  for (const commitHash of findCommitHashesForFile(filePath)) {
+    const translation = retrieveJSONForFileAtCommitHash(filePath, commitHash);
+    const translationKeys = Object.keys(translation).map(key => normalizeTranslationKey(key));
+    if (translationKeysByLocale[locale] === undefined) {
+      translationKeysByLocale[locale] = new Set([]);
+    }
+    translationKeysByLocale[locale] = new Set([...translationKeysByLocale[locale], ...translationKeys]);
+  }
+}
+
+/**
+ * Step 3: Add rows of missing translations to Google Spreadsheet.
+ * 
+ * https://docs.google.com/spreadsheets/d/1ILFfc1DX4ujMnLnf9UqhwQGM9Ke3s1cAWciy8VqMHZw
+ */
+(async () => {
+  const doc = new GoogleSpreadsheet('1ILFfc1DX4ujMnLnf9UqhwQGM9Ke3s1cAWciy8VqMHZw');
+  await doc.useServiceAccountAuth(require('./coronastatus-translation-486cef09736e-credentials.json'));
+  await doc.loadInfo();
+
+  const allLocales = retrieveAllLocales('app/locales/');
+  for (let sheetIndex = 0; sheetIndex < doc.sheetCount; sheetIndex++) {
+    const sheet = doc.sheetsByIndex[sheetIndex];
+    for (const locale of allLocales) {
+      if (sheet.title !== locale) {
```

Comment:
If I understand it correctly, there is one sheet per locale?

Is it then necessary to loop through the sheets here, can't we just loop through the locales and try to add them (from the code it seems like it throws an error if it does not exist)?
````
try {
          await doc.addSheet({ title: locale, headerValues: ['key', 'translation'] });
        } catch (error) {
          // We don't do anything if the sheet for this locale exists.
        }
````

Reply:
That's correct, take a look at the sheet here: https://docs.google.com/spreadsheets/d/1ILFfc1DX4ujMnLnf9UqhwQGM9Ke3s1cAWciy8VqMHZw/edit#gid=462835362

The reason I'm looping over locales is because that way we will create sheets for new locales automatically. There is no way in the API to retrieve the sheet names in the spreadsheet. This way we don't have to worry about keeping locales and sheets in sync (it will sort it self out).

Reply:
Didn't read your comment thoroughly enough. Yeah, maybe we can get away without that loop.

Reply:
Haha getting late... The reason we have to loop over those are because there is no way of retrieving a sheet by its name. Does that make sense?

Reply:
I see, I thought the for-loop ended after adding a sheet. A little hard to wrap my head around the logic, but as long as it works it is more than good enough.

Reply:
I agree. Just refactored it, better now?